pkg/agent/agent.go

```diff
@@ -1141,3 +1104,82 @@ func (i *Initializer) patchNodeAnnotations(nodeName, key string, value interface
 func (i *Initializer) getNodeInterfaceFromIP(nodeIPs *utilip.DualStackIPs) (v4IPNet *net.IPNet, v6IPNet *net.IPNet, iface *net.Interface, err error) {
 	return getIPNetDeviceFromIP(nodeIPs, sets.NewString(i.hostGateway))
 }
+
+func (i *Initializer) initLocalConfig() error {
+	nodeName, err := env.GetNodeName()
+	if err != nil {
+		return err
+	}
+	if !i.enableExternalNode {
+		err := i.initNodeLocalConfig(nodeName)
+		if err != nil {
+			return err
+		}
+		i.networkConfig.IPv4Enabled = config.IsIPv4Enabled(i.nodeConfig, i.networkConfig.TrafficEncapMode)
+		i.networkConfig.IPv6Enabled = config.IsIPv6Enabled(i.nodeConfig, i.networkConfig.TrafficEncapMode)
+		return nil
+	}
+	return i.initVMLocalConfig(nodeName)
+}
+
+func (i *Initializer) initVMLocalConfig(nodeName string) error {
+	i.nodeConfig = &config.NodeConfig{
+		Name:      nodeName,
+		Type:      config.VMNode,
+		OVSBridge: i.ovsBridge,
+	}
+	// Only IPv4 is required in ExternalNode feature.
+	i.networkConfig.IPv4Enabled = true
+	return nil
+}
+
+func (i *Initializer) installPodNodeInitialFlows() error {
```

Comment:
I think Hongliang's PR moves these flows to OVS pipeline init funcs.

Reply:
I forgot to rebase latest code, removed.